File: csharp/demo/demo12a-race-condition.cs
/*
 * RACE CONDITIONS
 */
using System;
using System.Collections.Generic;
using System.Threading;



class Demo12A : IRunnable
{
    public void run()
    {
        const int NUM_THREADS = 5;
        var lstTh = new List<Thread>();


        for (int i = 0; i < NUM_THREADS; ++i)
        {
            int ith = i;

            lstTh.Add(new Thread(() =>
            {
                Thread.Sleep(1000);
                Console.WriteLine(ith);
            }));
        }


        lstTh.ForEach(th => th.Start());
    }
}

File: python/demo12a_race_condition.py
'''
RACE CONDITIONS
'''

import time
import threading



def do_task(index):
    time.sleep(1)
    print(index, end='')



NUM_THREADS = 4
lstth = []

for i in range(NUM_THREADS):
    lstth.append(threading.Thread( target=do_task, args=(i,) ))

for th in lstth:
    th.start()
